pkg/controllermanager/controller/project/project.go

```diff
@@ -145,6 +159,52 @@ func (c *Controller) Run(ctx context.Context, workers int) {
 		}
 	}()
 
+	// TODO: This can be removed in a future version again.
+	// The following code assigns the 'uam' role to every project member having the role 'admin'. This is only done in
+	// case the UAM cluster role was not created yet, i.e., for existing projects. It won't happen for new projects.
+	{
+		logger.Logger.Info("Project controller user access management migration started.")
+		gardenClient, err := c.clientMap.GetClient(ctx, keys.ForGarden())
+		if err != nil {
+			panic(fmt.Errorf("failed to get garden client: %w", err))
+		}
+
+		projectList, err := c.projectLister.List(labels.Everything())
+		if err != nil {
+			panic(err)
+		}
+
+		for _, project := range projectList {
+			if _, err := c.clusterRoleLister.Get("gardener.cloud:system:project-uam:" + project.Name); err != nil {
+				if !apierrors.IsNotFound(err) {
+					panic(err)
+				}
+
+				var (
+					projectCopy = project.DeepCopy()
+					memberNames []string
+				)
+
+				for i, member := range projectCopy.Spec.Members {
+					if (member.Role == gardencorev1beta1.ProjectMemberAdmin || utils.ValueExists(gardencorev1beta1.ProjectMemberAdmin, member.Roles)) && !utils.ValueExists(gardencorev1beta1.ProjectMemberUserAccessManager, member.Roles) {
```

Comment:
This means that even service accounts will get the uam role.
If I am a human and somebody with `uam` role disallow me to manage human users, I can use one of the existing service accounts and escalate my privileges.

It will be similar to the current behavior, but maybe we would like to change this with the `uam` role?

Reply:
This is for backwards compatibility for existing resources (users, Service Accounts). The owner/admin will decide from whom to remove the uam role once we release the new change. For the new projects the uam role will not go together with the admin role. It will be assigned explicitly.